### DISCUSSION TITLE
Does iterating on features that require  replays usually involve downtime?

### DISCUSSION THREAD

**Discussion by davidharting** [hirethunk/verbs#176] [Q&A]
_There may be some bad assumptions built into this question! Please point that out if I'm being wrong-headed here._

I've read through the docs and listened to Daniel on podcasts, and drawn two conclusions that seem to contradict a little

1. Make models that are derived from `handle()` methods minimal. Just what the UI needs.
2. Replaying events will likely involve downtime (truncate `handled()`d model tables, and replay. Downtime while waiting for that).

For an event-sourced app with active users and active development, does this cause tension? 

If replaying events were free, you would want to do so as part of every deployment, just like migrations(right?). But they may involve downtime, so do you need to plan for when you replay events and fully release new features that involve updated projected models?

Curious what folks with production experience on Verbs have to say about this! Is this tension real?

**Comment by gdespirito** [comment]
I think that you can do your handle function idempotent, so it can be run live without downtime. I think that downtimes is only needed in case you have to truncate an entire table, and then replay the events to create that records again.

**Reply by davidharting** [reply]
Ah right - okay that makes sense. Idempotency as a solution here makes sense. In some cases probably easier said than done, but worth it for ease of reply in my case.
